spras/interactome.py

```diff
@@ -0,0 +1,123 @@
+"""
+Author: Neha Talluri
+07/19/23
+
+Methods for converting from the universal input and recreating the universal output
+"""
+
+import pandas as pd
+
+
+def convert_undirected_to_directed(df: pd.DataFrame) -> pd.DataFrame:
+    """
+    turns a graph into a fully directed graph
+    - turns every unidirected edges into a bi-directed edge
+    - with bi-directed edges, we are not loosing too much information because the relationship of the undirected egde is still preserved
+
+   * A user must keep the Direction column when using this function
+
+    @param df: input network df of edges, weights, and directionality
+    @return a dataframe with no undirected edges in Direction column
+    """
+
+    mask = df['Direction'] == 'U'
+    new_df = df[mask].copy(deep=True)
+    new_df['Interactor1'], new_df['Interactor2'] = new_df['Interactor2'], new_df['Interactor1']
+    new_df['Direction'] = 'D'
+    df.loc[mask, 'Direction'] = 'D'
+    df = pd.concat([df, new_df], ignore_index=True)
+    return df
+
+
+def convert_directed_to_undirected(df: pd.DataFrame) -> pd.DateOffset:
+    """
+    turns a graph into a fully undirected graph
+    - turns all the directed edges directly into undirected edges
+    - we will loose any sense of directionality and the graph won't be inherently accurate, but the basic relationship between the two connected nodes will still remain intact.
+
+    * A user must keep the Direction column when using this function
+
+    @param df: input network df of edges, weights, and directionality
+    @return a dataframe with no directed edges in Direction column
+    """
+
+    df["Direction"] = "U"
+
+    return df
+
+
+def add_constant(df: pd.DataFrame, new_col_name: str, const: str) -> pd.DataFrame:
+    """
+    adds a constant at the end of the input dataframe that is needed inbetween columns for a specifc algorithm
+
+    @param df: input network df of edges, weights, and directionality
+    @param new_col_name: the name of the new column
+    @param const: some type of constant needed in the df
+    @return a df with a new constant added to every row
+    """
+
+    df.insert(df.shape[1], new_col_name, const)
+
+    return df
+
+
+def add_directionality_constant(df: pd.DataFrame,  col_name: str, dir_const: str, undir_const: str) -> pd.DataFrame:
```

Comment:
Would `add_directionality_sep` be a better name for this? The value isn't a constant in this case because it is different for directed and undirected edges. I would then change the arguments to `dir_sep` and `undir_sep` as well. If we change them here, we should also change them in `readd_direction_col_mixed`.

It would likely help document these functions if this function and all of the above functions include something in the docstring about being used to generate inputs whereas the following functions are used for parse output.